src/JSInterop/Microsoft.JSInterop/src/IJSInteropTask.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+
+namespace Microsoft.JSInterop;
+
+internal interface IJSInteropTask : IDisposable
+{
+    public Type ResultType { get; }
+
+    public JsonSerializerOptions? DeserializeOptions { get; set; }
```

Comment:
Why specifically deserialization options?

Reply:
Maybe rename it to `SerializerOptions` so it doesn't appear that it's a different set of options.

Reply:
Since this is `internal` we probably don't need to block the PR on its naming. Mackinnon may prefer to rename this, or in my opinion it would be equally fine just to add a code comment here to clarify its purpose.